traitsui/qt4/list_editor.py

```diff
@@ -311,7 +311,12 @@ def add_item(self, offset):
         index += offset
         item_trait = self._trait_handler.item_trait
         value = item_trait.default_value_for(self.object, self.name)
-        self.value = list[:index] + [value] + list[index:]
+        try:
+            self.value = list[:index] + [value] + list[index:]
+        # if the default new item is invalid, we just don't add it to the list.
+        # traits will still give an error message, but we don't want to crash
+        except TraitError:
+            pass
```

Comment:
It feels not ideal to just swallow the exception and leaving the list as it was before, but that feels better than crashing entirely

Reply:
Yes, use `raise_to_debug`.  We should probably using that more often.